Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -281,11 +279,40 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
 
     // MARK: ApplePay
 
-    let paymentAuthorizationData: Signal<PostCampaignPaymentAuthorizationData, Never> = self
+    /*
+
+     Order of operations:
+     1) Apple pay button tapped
+     2) Generate a new payment intent
+     3) Present the payment authorization form
+     4) Payment authorization form calls applePayContextDidCreatePayment with ApplePay params
+     5) Payment authorization form calls paymentAuthorizationDidFinish
+     */
+
+    let newPaymentIntentForApplePay: Signal<String, Never> = self.configureWithDataProperty.signal
+      .skipNil()
+      .takeWhen(self.applePayButtonTappedSignal)
```

Comment:
As a fast follow, I'll create a ticket to make sure the Apple Pay button can't be tapped twice.